From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JustMangoT <38831897+JustMangoT@users.noreply.github.com>
Date: Mon, 17 Jan 2022 23:17:29 +0700
Subject: [PATCH] (AtelierMC) Ignore tab completions if permissions are missing

Original code by AtelierMC, licensed under MIT
You can find the original code on https://github.com/AtelierMC/AtelierMC

diff --git a/src/main/java/org/bukkit/command/PluginCommand.java b/src/main/java/org/bukkit/command/PluginCommand.java
index 7f153000eee0f563ee4f1e3c218aadb5b6a33a97..d4554fa10d2ea2e795b7aae129c2504827b3d16b 100644
--- a/src/main/java/org/bukkit/command/PluginCommand.java
+++ b/src/main/java/org/bukkit/command/PluginCommand.java
@@ -134,12 +134,16 @@ public final class PluginCommand extends Command implements PluginIdentifiableCo
 
         List<String> completions = null;
         try {
-            if (completer != null) {
-                completions = completer.onTabComplete(sender, this, alias, args);
-            }
-            if (completions == null && executor instanceof TabCompleter) {
-                completions = ((TabCompleter) executor).onTabComplete(sender, this, alias, args);
+            // AtelierMC start - Don't show tab completions if the player doesn't have permission
+            if (testPermission(sender)) {
+                if (completer != null) {
+                    completions = completer.onTabComplete(sender, this, alias, args);
+                }
+                if (completions == null && executor instanceof TabCompleter) {
+                    completions = ((TabCompleter) executor).onTabComplete(sender, this, alias, args);
+                }
             }
+            // AtelierMC end
         } catch (Throwable ex) {
             StringBuilder message = new StringBuilder();
             message.append("Unhandled exception during tab completion for command '/").append(alias).append(' ');
